src/main/java/at/hannibal2/skyhanni/features/inventory/craft/CraftableItemList.kt

```diff
@@ -0,0 +1,129 @@
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryOpenEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.CollectionUtils.addOrPut
+import at.hannibal2.skyhanni.utils.CollectionUtils.sortedDesc
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.itemName
+import at.hannibal2.skyhanni.utils.ItemUtils.itemNameWithoutColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems
+import at.hannibal2.skyhanni.utils.NEUItems.getPrice
+import at.hannibal2.skyhanni.utils.NEUItems.isVanillaItem
+import at.hannibal2.skyhanni.utils.PrimitiveItemStack.Companion.toPrimitiveStackOrNull
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import io.github.moulberry.notenoughupdates.recipes.CraftingRecipe
+import io.github.moulberry.notenoughupdates.recipes.NeuRecipe
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.math.floor
+
+class CraftableItemList {
+    private val config get() = SkyHanniMod.feature.inventory.craftableItemList
+
+    private var display = listOf<Renderable>()
+    private var inInventory = false
+    private val craftItemPattern by RepoPattern.pattern(
+        "craftableitemlist.craftitem",
+        "Craft Item"
+    )
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!isEnabled()) return
+        if (!craftItemPattern.matches(event.inventoryName)) return
+        inInventory = true
+
+        val pricePer = mutableMapOf<NEUInternalName, Double>()
+        val lines = mutableMapOf<NEUInternalName, Renderable>()
+        val avaliableMaterial = readItems()
+        for (internalName in NEUItems.allInternalNames) {
+            if (config.excludeVanillaItems && !internalName.isVanillaItem()) continue
+
+            val recipes = NEUItems.getRecipes(internalName)
+
+            for (recipe in recipes) {
+                if (recipe !is CraftingRecipe) continue
+                val neededItems = neededItems(recipe, internalName.toString())
+                // Just a fail save, should not happen normally
+                if (neededItems.isEmpty()) continue
+                val amount = canCraftAmount(neededItems, avaliableMaterial)
+                if (amount > 0) {
+                    pricePer[internalName] = pricePer(neededItems)
+                    lines[internalName] = Renderable.clickAndHover("§8x$amount ${internalName.itemName}",
+                        tips = listOf("Click to craft ${internalName.itemNameWithoutColor}!"),
```

Comment:
this feels plain without colour, i think it should have it

Reply:
agreed